test/smartsheet/api/retry_net_client_decorator_test.rb

```diff
@@ -40,36 +47,103 @@ def given_non_retryable_response
 
   it 'does not retry when a non-retryable failure occurs' do
     given_non_retryable_response
-    @client.expects(:make_request).returns @response
+    @client.expects(:make_request).with(REQUEST).returns @response
 
     Smartsheet::API::RetryNetClientDecorator
       .new(@client, Smartsheet::API::RetryLogic.new)
-      .make_request(
-        Smartsheet::API::Request.new(
-          'token',
-          Smartsheet::API::EndpointSpec.new(:get, ['sheets']),
-          Smartsheet::API::RequestSpec.new,
-          @base_url
-        )
-      )
+      .make_request(REQUEST)
   end
 
   it 'retries when a retryable failure occurs' do
     given_retryable_response
-    @client.expects(:make_request).at_least(2).returns @response
+    @client.expects(:make_request).with(REQUEST).at_least(2).returns @response
 
     retrier = Smartsheet::API::RetryLogic.new
     stub_sleep(retrier)
 
     Smartsheet::API::RetryNetClientDecorator
       .new(@client, retrier)
-      .make_request(
-        Smartsheet::API::Request.new(
-          'token',
-          Smartsheet::API::EndpointSpec.new(:get, ['sheets']),
-          Smartsheet::API::RequestSpec.new,
-          @base_url
-        )
-      )
+      .make_request(REQUEST)
+  end
+
+  it 'logs retry failure count after retrying more than once' do
+    given_retryable_response
+    @client.stubs(:make_request).with(REQUEST).returns @response
+
+    retrier = Smartsheet::API::RetryLogic.new
+    stub_sleep(retrier)
+
+    logger = mock
+    logger.expects(:log_retry_failure).with { |x| x >= 2 }
+    logger.stubs(:log_retry_attempt)
+
+    Smartsheet::API::RetryNetClientDecorator
+        .new(@client, retrier, logger: logger)
+        .make_request(REQUEST)
+  end
+
+  it 'logs retry attempts when a retryable failure occurs' do
+    given_retryable_response
+    @client.stubs(:make_request).with(REQUEST).returns @response
```

Comment:
You could probably pull this line into some sort of base given_*_response method. It seems to appear in each test